File: Hashing/python/pairSum.py
from typing import List, Tuple


def pair_sum(numbers: List[int], target_sum: int) -> Tuple[int, int]:
    indices: dict[int, int] = {}
    for idx, value in enumerate(numbers):
        complement = target_sum - value
        if complement in indices:
            return (indices[complement], idx)
        indices[value] = idx
    raise ValueError("Pair not found")

File: Hashing/php/pairSum.php
<?php

declare(strict_types=1);

function pairSum(array $numbers, int $targetSum): array
{
    $indices = [];
    foreach ($numbers as $idx => $value) {
        $complement = $targetSum - $value;
        if (array_key_exists($complement, $indices)) {
            return [$indices[$complement], $idx];
        }
        if (!array_key_exists($value, $indices)) {
            $indices[$value] = $idx;
        }
    }

    throw new InvalidArgumentException('Pair not found');
}
